src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -106,6 +108,8 @@ public class ServerStore : IDisposable
 
         public Operations Operations { get; }
 
+        private readonly ConcurrentDictionary<string, long > _lastRaftIndexByDatabase = new ConcurrentDictionary<string, long>();
```

Comment:
Why not simply load the database record and check its etag?

Reply:
to save the last raft index for compare exchange and identities